cmd/snap-update-ns/utils.go

```diff
@@ -62,10 +66,92 @@ type ReadOnlyFsError struct {
 	Path string
 }
 
+// Error returns a formatted error message.
 func (e *ReadOnlyFsError) Error() string {
 	return fmt.Sprintf("cannot operate on read-only filesystem at %s", e.Path)
 }
 
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *ReadOnlyFsError) MimicPath() string {
+	return e.Path
+}
+
+// TrespassingError is an error when filesystem operation would affect the host.
+type TrespassingError struct {
+	Path string
+}
+
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *TrespassingError) Error() string {
+	return fmt.Sprintf("cannot trespass on host filesystem at %s", e.Path)
+}
+
+// MimicPath returns the path of the directory where a mimic ought to be constructed.
+func (e *TrespassingError) MimicPath() string {
+	return e.Path
+}
+
+// TODO: Handle /home/*/snap/* when we do per-user mount namespaces and allow
+// defining layout items that refer to SNAP_USER_DATA and SNAP_USER_COMMON.
+//
+// Note that /snap is "not allowed" because it is handled by existing EROFS code.
```

Comment:
And then main changes it?

Reply:
The trick is that /snap is actually on a writable medium so we need a tailored rule to allow creating things under /snap/$SNAP_NAME/ or the more strict check for filesystem type kicks in. I will adjust the comment to explain this.

Reply:
This is fixed now